community/kernel/src/main/java/org/neo4j/kernel/impl/api/ConstraintEnforcingEntityOperations.java

```diff
@@ -99,39 +98,50 @@ public boolean nodeAddLabel( KernelStatement state, NodeItem node, int labelId )
         {
             PropertyConstraint constraint = constraints.next();
             int propertyKeyId = constraint.propertyKey();
-            Object propertyValue = node.getProperty( propertyKeyId );
-            if ( propertyValue != null )
+            try ( Cursor<NodeItem> cursor = loadNodeById( state, nodeId ) )
             {
-                validateNoExistingNodeWithLabelAndProperty( state, labelId, propertyKeyId, propertyValue, node.id() );
+                NodeItem node = cursor.get();
+                Object propertyValue = node.getProperty( propertyKeyId );
+                if ( propertyValue != null )
+                {
+                    validateNoExistingNodeWithLabelAndProperty( state, labelId, propertyKeyId, propertyValue, node.id() );
+                }
             }
 
         }
-        return entityWriteOperations.nodeAddLabel( state, node, labelId );
+        return entityWriteOperations.nodeAddLabel( state, nodeId, labelId );
     }
 
     @Override
-    public Property nodeSetProperty( KernelStatement state, NodeItem node, DefinedProperty property )
-            throws ConstraintValidationKernelException
+    public Property nodeSetProperty( KernelStatement state, long nodeId, DefinedProperty property )
+            throws ConstraintValidationKernelException, EntityNotFoundException
     {
-        try ( Cursor<LabelItem> labels = node.labels() )
+        try ( Cursor<NodeItem> cursor = loadNodeById( state, nodeId ) )
         {
-            while ( labels.next() )
+
+            NodeItem node = cursor.get();
+
+            try ( Cursor<LabelItem> labels = node.labels() )
             {
-                int labelId = labels.get().getAsInt();
-                int propertyKeyId = property.propertyKeyId();
-                Iterator<NodePropertyConstraint> constraintIterator =
-                        uniquePropertyConstraints(
-                                schemaReadOperations.constraintsGetForLabelAndPropertyKey( state, labelId,
-                                        propertyKeyId ) );
-                if ( constraintIterator.hasNext() )
+                while ( labels.next() )
                 {
-                    validateNoExistingNodeWithLabelAndProperty( state, labelId, property.propertyKeyId(),
-                            property.value(), node.id() );
+                    int labelId = labels.get().getAsInt();
+                    int propertyKeyId = property.propertyKeyId();
+                    Iterator<NodePropertyConstraint> constraintIterator =
+                            uniquePropertyConstraints(
+                                    schemaReadOperations.constraintsGetForLabelAndPropertyKey( state, labelId,
+                                            propertyKeyId ) );
+                    if ( constraintIterator.hasNext() )
+                    {
+                        validateNoExistingNodeWithLabelAndProperty(
```

Comment:
well this is not something related to this PR, so I wouldn't mind to deal with it in a different PR